lib/winapi/windef.h

```diff
@@ -0,0 +1,21 @@
+#ifndef __WINDEF_H__
+#define __WINDEF_H__
+
+typedef const void *LPCVOID;
+typedef void VOID, *PVOID, *LPVOID;
+typedef PVOID HANDLE, *PHANDLE;
+
+typedef char CHAR;
+typedef long LONG, *PLONG;
+typedef long long LONGLONG, *PLONGLONG;
+
+typedef unsigned short WORD, WCHAR, *PWSTR;
+typedef unsigned long ULONG, *PULONG;
+typedef unsigned int DWORD, *PDWORD, *LPDWORD;
+
+typedef unsigned int SIZE_T;
+
+typedef int BOOL, *PBOOL;
+typedef const char *PCSZ, *PCSTR, *LPCSTR;
+
+#endif
```

Comment:
I believe these `typedef` also exist in xboxkrnl.h? Shouldn't they be removed from there?

I'm surprised you didn't get a warning or even errors.

Reply:
Redundant typedefs seem to be fine in C (at least in clang), as long as they're all identical.
My intention for duplicating them was to keep xboxkrnl.h independent from my WinAPI work, especially since I didn't know what else might eventually end up in windef.h (and therefore pollute the namespace) - I added the types on an as-needed basis.
If de-duplicating them is considered cleaner, I can add that change though.

Reply:
I think the cleanest approach (and my personal preference) is to have these types defined in one place.

Reply:
I added another commit that splits out the typdefs into `xboxdef.h`, which gets included from both `xboxkrnl.h` and `windef.h` (I didn't name the new header `windef.h` as suggested on Discord to avoid include guard conflicts etc.).